ydb/core/persqueue/pq_impl.cpp

```diff
@@ -2345,6 +2464,94 @@ void TPersQueue::HandleSplitMessageGroupRequest(ui64 responseCookie, const TActo
     ctx.Send(partActor, new TEvPQ::TEvSplitMessageGroup(responseCookie, std::move(deregistrations), std::move(registrations)));
 }
 
+void TPersQueue::HandleGetOwnershipRequestForShadowPartition(const ui64 responseCookie,
+                                                             const NKikimrClient::TPersQueuePartitionRequest& req,
+                                                             const TActorId& sender,
+                                                             const TActorContext& ctx)
+{
+    Y_ABORT_UNLESS(req.HasWriteId());
+
+    ui64 writeId = req.GetWriteId();
+    ui32 partitionId = req.GetPartition();
+
+    if (TxWrites.contains(writeId) && TxWrites[writeId].Partitions.contains(partitionId)) {
+        ui32 shadowPartitionId = TxWrites[writeId].Partitions[partitionId];
+        Y_ABORT_UNLESS(ShadowPartitions.contains(shadowPartitionId));
+        TPartitionInfo& partition = ShadowPartitions.at(shadowPartitionId);
+
+        if (partition.InitDone) {
+            const TActorId& actorId = partition.Actor;
+            TActorId pipeClient = ActorIdFromProto(req.GetPipeClient());
+
+            HandleGetOwnershipRequest(responseCookie, actorId, req, ctx, pipeClient, sender);
+        } else {
+            partition.GetOwnershipRequests.emplace_back(responseCookie, req, sender);
+        }
+    } else {
+        GetOwnershipRequests.push_back({responseCookie, req, sender});
+
+        TryWriteTxs(ctx);
+    }
+}
+
+void TPersQueue::HandleReserveBytesRequestForShadowPartition(const ui64 responseCookie,
+                                                             const NKikimrClient::TPersQueuePartitionRequest& req,
+                                                             const TActorId& sender,
+                                                             const TActorContext& ctx)
+{
+    Y_ABORT_UNLESS(req.HasWriteId());
+
+    ui64 writeId = req.GetWriteId();
+    ui32 partitionId = req.GetPartition();
+
+    Y_ABORT_UNLESS(TxWrites.contains(writeId) && TxWrites[writeId].Partitions.contains(partitionId));
```

Comment:
TODO: use the same logic in all 3 handlers. Create partition if needed, store event in queue. And process it only when shadow partition initied.

Reply:
done
https://github.com/ydb-platform/ydb/pull/1037/commits/d2b1192e2d969c93ca2b18e033bcbd38425aeecd